File: docs/breez-sdk/snippets/swift/BreezSdkSnippets/Sources/WebLn.swift
import BreezSdkSpark
import WebKit

// ANCHOR: webln-integration
func setupWebLn(sdk: BreezSdk, webView: WKWebView) {
    // Create the WebLN controller with the SDK and WebView
    let controller = WebLnController(
        sdk: sdk,
        webView: webView,
        onEnableRequest: { domain in
            // Show a dialog asking the user to approve WebLN access
            // Return true to allow, false to deny
            await showEnableDialog(domain: domain)
        },
        onPaymentRequest: { invoice, amountSats in
            // Show a dialog asking the user to approve the payment
            // Return true to approve, false to reject
            await showPaymentDialog(invoice: invoice, amountSats: amountSats)
        },
        onLnurlRequest: { request in
            // Handle LNURL requests (pay, withdraw, auth)
            switch request.type {
            case .pay:
                // Show UI to select amount within min/max bounds
                // Return LnurlUserResponse with approved, amountSats, and optional comment
                return LnurlUserResponse(approved: true, amountSats: 1000)
            case .withdraw:
                // Show UI to select amount within min/max bounds
                return LnurlUserResponse(approved: true, amountSats: 1000)
            case .auth:
                // Show confirmation dialog
                return LnurlUserResponse(approved: true)
            }
        }
    )

    // Inject the WebLN provider into the WebView
    controller.inject()
}
// ANCHOR_END: webln-integration

// Placeholder functions - implement these with your UI framework
func showEnableDialog(domain: String) async -> Bool {
    // Show your own permission dialog here
    return true
}

func showPaymentDialog(invoice: String, amountSats: Int64) async -> Bool {
    // Show your own payment confirmation dialog here
    return true
}

File: docs/breez-sdk/snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/WebLn.kt
package com.example.kotlinmpplib

import breez_sdk_spark.*
import breez_sdk_spark.webln.*

class WebLn {
    // ANCHOR: webln-integration
    // On Android: val webView = WebLnWebView.fromAndroid(androidWebView)
    // On iOS: val webView = WebLnWebView.fromIos(wkWebView)
    suspend fun setupWebLn(sdk: BreezSdk, webView: WebLnWebView) {
        // Create the WebLN controller with the SDK and WebView wrapper
        val controller = WebLnController(
            sdk = sdk,
            webView = webView,
            onEnableRequest = { domain ->
                // Show a dialog asking the user to approve WebLN access
                // Return true to allow, false to deny
                showEnableDialog(domain)
            },
            onPaymentRequest = { invoice, amountSats ->
                // Show a dialog asking the user to approve the payment
                // Return true to approve, false to reject
                showPaymentDialog(invoice, amountSats)
            },
            onLnurlRequest = { request ->
                // Handle LNURL requests (pay, withdraw, auth)
                when (request.type) {
                    LnurlType.PAY -> {
                        // Show UI to select amount within min/max bounds
                        // Return LnurlUserResponse with approved, amountSats, and optional comment
                        LnurlUserResponse(approved = true, amountSats = 1000L)
                    }
                    LnurlType.WITHDRAW -> {
                        // Show UI to select amount within min/max bounds
                        LnurlUserResponse(approved = true, amountSats = 1000L)
                    }
                    LnurlType.AUTH -> {
                        // Show confirmation dialog
                        LnurlUserResponse(approved = true)
                    }
                }
            }
        )

        // Inject the WebLN provider into the WebView
        controller.inject()
    }

    // When done, clean up resources
    fun cleanup(controller: WebLnController) {
        controller.dispose()
    }
    // ANCHOR_END: webln-integration

    // Placeholder functions - implement these with your UI framework
    private suspend fun showEnableDialog(domain: String): Boolean {
        // Show your own permission dialog here
        return true
    }

    private suspend fun showPaymentDialog(invoice: String, amountSats: Long): Boolean {
        // Show your own payment confirmation dialog here
        return true
    }
}
